Nos/Models/Event+CoreDataClass.swift

```diff
@@ -805,6 +806,77 @@ public class Event: NosManagedObject {
         return nil
     }
     
+    // MARK: - Preloading and Caching
+    // Probably should refactor this stuff into a view model
```

Comment:
I want to refactor this too but there are a few reasons I didn't want to do it in this PR:
- It will make the diff huge because every view that holds a reference to a `Event` will need to change
- I'm not sure how to make this work with `@FetchRequest` which returns NSManagedObjects and notifies us when they change. If we used a view model we would need to wrap @FetchRequest or something to instantiate view models when new objects are inserted? Same with NSFetchedResultsController. We could make a separate object that manages this cache of data but then we have to figure out cache invalidation.

I opened [#764](https://github.com/planetary-social/nos/issues/764) to come back and figure this out later.